packages/@aws-cdk/custom-resources/lib/provider-framework/provider.ts

```diff
@@ -0,0 +1,198 @@
+import iam = require('@aws-cdk/aws-iam');
+import lambda = require('@aws-cdk/aws-lambda');
+import sfn = require('@aws-cdk/aws-stepfunctions');
+import tasks = require('@aws-cdk/aws-stepfunctions-tasks');
+import { Construct, Duration } from '@aws-cdk/core';
+import path = require('path');
+import consts = require('./runtime/consts');
+import { calculateRetryPolicy } from './util';
+
+const RUNTIME_HANDLER_PATH = path.join(__dirname, 'runtime');
+
+/**
+ * Initialization properties for the `Provider` construct.
+ */
+export interface ProviderProps {
```

Comment:
`timeout` and `memorySize` for the handler are missing and then there is maybe some validation logic to do with `timeout` and `totalTimeout`.

And `onEventHandler`/`isCompleteHandler` must have timeouts greater than this `timeout` because they invoke the user handlers synchronously.

Reply:
Good point. I guess other AWS Lambda options such as setting environment variables should also be exposed.
Do you think it might make more sense to simply pass in two `lambda.IFunction` constructs instead?

Reply:
And then a `timeout` of 15 minutes for the `onEventFunction`/`isCompleteFunction` because you don't know the timeout of the user functions...?

Also the total invocation time of the user functions cannot exceed 15 minutes because they can both be invoked synchronously in the `onEventFunction`.

Reply:
yes I guess 15 minutes for both functions is the max. One thing we could do is not invoke isComplete synchronously which will clean this up a little. I’ve since changed this a bit such that if isComplete is not defined, the provider is synchronous so the value of the immediate sync call is reduced.

Reply:
Set all framework timeouts to 15 minutes for now. I think this should be fine.

Reply:
Where? `timeout` needs to be set to 15 minutes in `private createFunction(entrypoint: string)`, no? You don't want your function to timeout before the user function

Reply:
See latest version and let me know if it makes sense